### PR TITLE
add swarm label of provider by default

### PR DESCRIPTION
**Pull request by ehazlett**
This will add a default label of `provider=<provider>` when using Swarm.

This enables things like `docker run -e constraint:provider==digitalocean ...`

### CODE REVIEW

File: host.go [hunk 1]
```diff
@@ -322,16 +322,19 @@ func (h *Host) generateDockerConfig(dockerPort int, caCertPath string, serverKey
...
-	// TODO @ehazlett: template?
+	if h.SwarmHost != "" {
+		swarmLabels = append(swarmLabels, fmt.Sprintf("--label=provider=%s", h.Driver.DriverName()))
+	}
+
 	defaultDaemonOpts := fmt.Sprintf(`--tlsverify \
 --tlscacert=%s \
 --tlskey=%s \
---tlscert=%s`, caCertPath, serverKeyPath, serverCertPath)
+--tlscert=%s %s`, caCertPath, serverKeyPath, serverCertPath, strings.Join(swarmLabels, " "))
```
Comment (sthulb) [inline host.go]: Not sure of the formatting for this line, I'd perhaps like to see:
```go

	defaultDaemonOpts := fmt.Sprintf(
		"--tlsverify --tlscacert=%s --tlskey=%s --tlscert=%s %s`, 
		caCertPath,
		serverKeyPath,
		serverCertPath, 
		strings.Join(swarmLabels, " ")
	)
```

File: host.go [hunk 2]
```diff
@@ -322,16 +322,19 @@ func (h *Host) generateDockerConfig(dockerPort int, caCertPath string, serverKey
 		daemonOpts    string
 		daemonOptsCfg string
 		daemonCfg     string
+		swarmLabels   = []string{}
 	)
 
-	// TODO @ehazlett: template?
+	if h.SwarmHost != "" {
```
Comment (vieux) [inline host.go]: Why only for swarmhost ? I would do it on every host, so if later it become a swarm host, no problem.

Comment (ehazlett) [inline host.go]: @vieux yes sir -- that makes more sense :)  thx!

File: host.go (hunk 1 above)
Comment (ehazlett) [inline host.go]: sure -- the reason i went multi-line is i was anticipating more engine options being added.  if we go with cloudinit it won't be there anyway.  i can update it - it should evaluate the same.

**Comment by ehazlett** [comment]
@sthulb updated ptal

**Comment by vieux** [comment]
LGTM

**Pull request merged**